### ISSUE TITLE
Fapi Policy Signed out of memory for object contexts

### ISSUE THREAD

**Issue by williamcroberts** [tpm2-software/tpm2-pytss#168]
With swtpm it fails with:
```
======================================================== FAILURES ========================================================
______________________________________________ TestFapi.test_policy_signed _______________________________________________
[gw3] linux -- Python 3.8.5 /usr/bin/python3

self = <test.test_fapi.TestFapi object at 0x7faf3a2f5400>
cryptography_key = (<cryptography.hazmat.backends.openssl.ec._EllipticCurvePrivateKey object at 0x7faf3a2f5430>, '-----BEGIN PUBLIC KEY--...fWsTUMIoGX3PTrgtbReITu8S+uw\nnkkC6JxCx4501TFBhHuDXcciPHkOaIYUxgdvylULrjszNxeIaN+1cQBeHA==\n-----END PUBLIC KEY-----\n')

    @pytest.mark.skipif(
        pkgconfig.installed("tss2-fapi", "<3.1.0"), reason="tpm2-tss bug, see #2080"
    )
    def test_policy_signed(self, cryptography_key):
        # create external signing key used by the signing authority external to the TPM
        sign_key, sign_key_public_pem = cryptography_key
    
        # create policy Signed, which is satisfied via a signature by sign_key
        policy = f"""
        {{
            "description": "Description pol_signed",
            "policy": [
                {{
                    "type": "PolicySigned",
                    "publicKeyHint": "Test key hint",
                    "keyPEM": "{sign_key_public_pem}",
                }}
            ]
        }}
        """
        policy_path = f"/policy/policy_{random_uid()}"
        self.fapi.import_object(path=policy_path, import_data=policy)
    
        # create key which can only be used if policy Signed is satisfied
        profile_name = self.fapi.config.profile_name
        key_path = f"/{profile_name}/HS/SRK/key_{random_uid()}"
>       self.fapi.create_key(path=key_path, type="sign", policy_path=policy_path)

test/test_fapi.py:735: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tpm2_pytss/FAPI.py:299: in create_key
    _chkrc(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

rc = 2306, acceptable = None

    def _chkrc(rc, acceptable=None):
        if rc == lib.TPM2_RC_SUCCESS:
            return
        if acceptable is not None and (rc == acceptable or rc in acceptable):
            return
>       raise TSS2_Exception(rc)
E       tpm2_pytss.TSS2_Exception.TSS2_Exception: tpm:warn(2.0): out of memory for object contexts

tpm2_pytss/utils.py:19: TSS2_Exception
-------------------------------------------------- Captured stderr call --------------------------------------------------
WARNING:esys:src/tss2-esys/api/Esys_Create.c:398:Esys_Create_Finish() Received TPM Error 
ERROR:fapi:src/tss2-fapi/fapi_util.c:3381:ifapi_key_create() ErrorCode (0x00000902) Key create finish 
ERROR:fapi:src/tss2-fapi/api/Fapi_CreateKey.c:283:Fapi_CreateKey_Finish() Key create ErrorCode (0x00000902) 
ERROR:fapi:src/tss2-fapi/api/Fapi_CreateKey.c:116:Fapi_CreateKey() ErrorCode (0x00000902) Key_Create
```

**Comment by williamcroberts** [comment]
@joholl

**Comment by joholl** [comment]
TL;DR you need a bleeding edge TSS

This is a known problem. The issue is that Fapi_Quote() leaves transient handles behind if a policy check fails. So earlier tests leave garbage behind. This was tracked in tpm2-software/tpm2-tss/issues/2084 and is fixed already.

**Comment by williamcroberts** [comment]
@joholl does this suffer from the same version checking issue in #169?

**Comment by joholl** [comment]
Yes, I think so. This is also fixed in master.

**Comment by williamcroberts** [comment]
> Yes, I think so. This is also fixed in master.

By master you mean tpm2-tss or this project?

**Comment by williamcroberts** [comment]
We got these all fixed IIUC.
